ui/admin/app/components/form/policy/policy-selection/index.js

```diff
@@ -13,41 +13,34 @@ export default class FormPolicySelectionComponent extends Component {
    * @type {boolean}
    */
   get showCustomInput() {
-    if (this.args.name === 'retention_policy') {
-      return this.isCustomRetentionSelected;
-    } else if (
-      this.args.name === 'deletion_policy' &&
-      this.args.model.retain_for?.days >= 0
-    ) {
-      return this.isCustomDeletionSelected;
-    } else {
-      return null;
-    }
+    return this.args.selectedOption === 'custom' ? true : false;
   }
 
   /**
-   * Returns true if rentention days are greater than 0
+   * Disable toggle when the form is disabled and when the retain/delete after days are 0
    * @type {boolean}
    */
-  get isCustomRetentionSelected() {
-    //no need to show custom input for the below days
-    const arr = [-1, 0, 2555, 2190];
-    return arr.includes(this.args.model.retain_for?.days) ? false : true;
+  get toggleDisabled() {
+    if (
+      this.args.disabled ||
+      this.args.model[this.args.customInputName]?.days === 0 ||
+      !this.args.model[this.args.customInputName]?.days
```

Comment:
perhaps this if statement can be simplified to just `this.args.disabled || !this.args.model[this.args.customInputName]?.days`

Reply:
hmm, I can't do that here though, because we want to disable the overrides in the UI when we select `do_not_protect` or `do_not_delete` and the API doesn't send me 0 days, it doesn't send anything if the days are 0. So I wanted to capture both behaviors here.

Reply:
Why is that? Isn't 0 already covered as a falsy value?

Reply:
I forgot 0 was a falsey value, fixed. ty both!